adapters/socketio_berkeley.c

```diff
@@ -910,32 +910,18 @@ int socketio_send(CONCRETE_IO_HANDLE socket_io, const void* buffer, size_t size,
                 ssize_t send_result = send(socket_io_instance->socket, buffer, size, 0);
                 if ((send_result < 0) || ((size_t)send_result != size))
                 {
-                    if (send_result == INVALID_SOCKET)
+                    if (errno != EAGAIN)
```

Comment:
How can we add a test for this?

Reply:
It would require an E2E test for it. We do not have E2E test modules specific for socketio_*. They do get tested indirectly through other E2E tests, but non would exercise this particular scenario.